File: back/src/main/java/com/back/domain/queue/dto/QueueEventResponse.java
package com.back.domain.queue.dto;

import com.back.domain.queue.constant.QueueEventType;

public record QueueEventResponse<T>(
        QueueEventType eventType,
        T data
) {
    public static <T> QueueEventResponse<T> of(QueueEventType eventType, T data) {
        return new QueueEventResponse<>(eventType, data);
    }
}

File: back/src/main/kotlin/com/back/domain/queue/dto/QueueEventResponse.kt
package com.back.domain.queue.dto

import com.back.domain.queue.constant.QueueEventType

data class QueueEventResponse<T>(
    val eventType: QueueEventType,
    val data: T
) {
    companion object {
        fun <T> of(eventType: QueueEventType, data: T): QueueEventResponse<T> {
            return QueueEventResponse(eventType, data)
        }
    }
}
